ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js

```diff
@@ -54,11 +57,17 @@ export function saveToCatalogModalDirective($rootScope, $uibModal) {
         link: link
     };
 
-    function link($scope) {
+    function link($scope, $element) {
         $scope.buttonText = $scope.config.label || ($scope.config.itemType ? `Update ${$scope.config.name || $scope.config.symbolicName}` : 'Add to catalog');
+
+        $injector.get('$templateCache').put('catalog-saver.modal.template.html', modalTemplate);
```

Comment:
You know that you can inject directly `$templateCache` without using the `$injector` service? Might be cleaner as you don't reuse it.